content/blog/generating-self-signed-certificates-using-cert-manager-for-kubernetes-in-hpe-greenlake-for-private-cloud-entreprise.md

```diff
@@ -0,0 +1,392 @@
+
+H﻿ere is a self-signed issuer YAML manifest file *issuer-selfsigned.yaml*:
+
+```shell
+$ cat issuer-selfsigned.yaml                                                            
+apiVersion: cert-manager.io/v1                                                                                                                   
+kind: Issuer                                                                                                                                     
+metadata:                                                                                                                                         
+ name: cfe-selfsigned-issuer                                                                                                                     
+spec:                                                                                                                                             
+ selfSigned: {}
+```
+
+Type the following commands to create a namespace in which you want to generate certificates and deploy the issuer. Replace the sample namespace *cfe-apps* in the commands with your own namespace.
+
+```shell
+$ kubectl create ns cfe-apps
+namespace/cfe-apps created
+
+$ kubectl apply -f issuer-selfsigned.yaml -n cfe-apps
+issuer.cert-manager.io/cfe-selfsigned-issuer created
+```
+
+T﻿ype the following command to check the deployed issuer in the namespace. The issuer should show *READY* as ***True***.
+
+```shell
+$ kubectl get issuer -n cfe-apps
+NAME                    READY   AGE
+
+cfe-selfsigned-issuer   True    7s
+```
+
+If you want to be able to request certificates from any namespace in a cluster, use the CRD resource called *ClusterIssuer*. 
+
+Here is a sample *ClusterIssuer* YAM manifest file *clusterissuer.yaml*: 
+
+```shell
+$ cat clusterissuer.yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-cluster-issuer
+spec:
+  selfSigned: {}
+```
+
+#### Generate a certificate
+
+Y﻿ou can use the CRD resource *Certificate* to generate a self-signed certificate. 
+
+H﻿ere is a sample *Certificate* YAML manifest file *certificate.yaml* :
+
+```shell
+$ cat certificate.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+ name: cfe-selfsigned-tls
+spec:
+ secretName: cfe-tls-key-pair
+ isCA: true
+ issuerRef:
+   name: cfe-selfsigned-issuer
+   kind: Issuer
+ commonName: "example.com"
+ dnsNames:
+ - nginx.example.com
+ - example.com
+```
+
+I﻿n this YAML file, the *commonName* is set to a sample domain *'example.com'*. The *dnsNames* includes *'example.com'* and its subdomain *'nginx.example.com'*. 
+
+Cert-manager supports generate wildcard certificates, e.g., using *'*.example.com'*, which allows to secure multiple subdomains under a single certificate. Wildcard certificates cover all subdomains under the specified domain. You need to be cautious when using them, as they grant access to any subdomain matching the pattern.
```

Comment:
The preview does not correctly rendered the wildcard for *.example.com.
Try to escape the special character (* in front of .example.com) using a backslash (\\) as shown below (you will need to switch to "Markdown" mode in the CMS editor):

*'\\\*.example.com'\*

This should then be rendered like that:  *'\*.example.com'*

Reply:
It's strange. I can see everything is Ok in the CMS editor. The *'\*.example.com'* is showing up in the source code. 

Well, I will change this wildcard domain without making it in italic.